src/pages/workspaces/workspace/Data.js

```diff
@@ -702,59 +719,61 @@ class EntitiesContent extends Component {
-      entityKey, entityMetadata, loadMetadata, firstRender
+      entityKey, entityMetadata, snapshotDetails, loadMetadata, firstRender
     } = this.props
     const { selectedEntities, deletingEntities, copyingEntities, refreshKey, showToolSelector, igvData: { selectedFiles, refGenome } } = this.state
-
     const { initialX, initialY } = firstRender ? StateHistory.get() : {}
     const selectedKeys = _.keys(selectedEntities)
     const selectedLength = selectedKeys.length
 
-    return selectedFiles ?
-      h(IGVBrowser, { selectedFiles, refGenome, workspace, onDismiss: () => this.setState(_.set(['igvData', 'selectedFiles'], undefined)) }) :
-      h(Fragment, [
-        h(DataTable, {
-          persist: true, firstRender, refreshKey, editable: !Utils.editWorkspaceError(workspace),
-          entityType: entityKey, entityMetadata, columnDefaults, workspaceId: { namespace, name },
-          onScroll: saveScroll, initialX, initialY,
-          selectionModel: {
-            selected: selectedEntities,
-            setSelected: e => this.setState({ selectedEntities: e })
-          },
-          childrenBefore: ({ entities, columnSettings }) => div({
-            style: { display: 'flex', alignItems: 'center', flex: 'none' }
-          }, [
-            this.renderDownloadButton(columnSettings),
-            !_.endsWith('_set', entityKey) && this.renderCopyButton(entities, columnSettings),
-            div({ style: { margin: '0 1.5rem', height: '100%', borderLeft: Style.standardLine } }),
-            div({ style: { marginRight: '0.5rem' } }, [`${selectedLength} row${selectedLength === 1 ? '' : 's'} selected`]),
-            this.renderSelectedRowsMenu(columnSettings)
-          ])
-        }),
-        deletingEntities && h(EntityDeleter, {
-          onDismiss: () => this.setState({ deletingEntities: false }),
-          onSuccess: () => {
-            this.setState({ deletingEntities: false, selectedEntities: {}, refreshKey: refreshKey + 1 })
-            loadMetadata()
-          },
-          namespace, name,
-          selectedEntities: selectedKeys, selectedDataType: entityKey, runningSubmissionsCount
-        }),
-        copyingEntities && h(ExportDataModal, {
-          onDismiss: () => this.setState({ copyingEntities: false }),
-          workspace,
-          selectedEntities: selectedKeys, selectedDataType: entityKey, runningSubmissionsCount
-        }),
-        h(ToolDrawer, {
-          workspace,
-          isOpen: showToolSelector,
-          onDismiss: () => this.setState({ showToolSelector: false }),
-          onIgvSuccess: newIgvData => this.setState({ showToolSelector: false, igvData: newIgvData }),
-          entityMetadata,
-          entityKey,
-          selectedEntities
-        })
-      ])
+    return Utils.cond(
+      [_.isArray(entityKey), ((snapshotDetails !== undefined) ? h(SnapshotInfo, { snapshotId: snapshotDetails[entityKey[0]].id } ) : spinnerOverlay())],
+      [selectedFiles ?
```

Comment:
Why not make `selectedFiles` it's own cond branch, rather than having this ternary here? Also, remember that the outcomes for a cond should be functions.